java/org/apache/tomcat/websocket/WsSession.java

```diff
@@ -805,8 +807,13 @@ protected MessageHandler getBinaryMessageHandler() {
     }
 
 
-    protected void updateLastActive() {
+    protected void updateLastActive(boolean isWriteOperation) {
         lastActive = System.currentTimeMillis();
+        if (isWriteOperation) {
+            lastWriteTime = lastActive;
+        } else {
+            lastReadTime = lastActive;
+        }
```

Comment:
On reflection I decided to split this into two methods for clarity when reading the calling code.